tests/end-to-end/e2e/testctx.go

```diff
@@ -0,0 +1,118 @@
+package e2e
+
+import (
+	"context"
+	"os"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/kyma-incubator/compass/tests/end-to-end/pkg/common"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/avast/retry-go"
+
+	"github.com/kyma-incubator/compass/tests/end-to-end/pkg/jwtbuilder"
+	gcli "github.com/machinebox/graphql"
+	"github.com/pkg/errors"
+)
+
+const defaultTenant = "2a1502ba-aded-11e9-a2a3-2a2ae2dbcce4"
+
+var tc *TestContext
+
+func init() {
+	var err error
+	tc, err = NewTestContext()
+	if err != nil {
+		panic(errors.Wrap(err, "while test context setup"))
+	}
+}
+
+// TestContext contains dependencies that help executing tests
+type TestContext struct {
+	Graphqlizer       common.Graphqlizer
+	gqlFieldsProvider common.GqlFieldsProvider
+	currentScopes     []string
+	cli               *gcli.Client
+}
+
+func NewTestContext() (*TestContext, error) {
+	scopesStr := os.Getenv("ALL_SCOPES")
+
+	currentScopes := strings.Split(scopesStr, " ")
+
+	bearerToken, err := jwtbuilder.Do(defaultTenant, currentScopes)
+	if err != nil {
+		return nil, errors.Wrap(err, "while building JWT token")
+	}
+
+	return &TestContext{
+		Graphqlizer:       common.Graphqlizer{},
+		gqlFieldsProvider: common.GqlFieldsProvider{},
+		currentScopes:     currentScopes,
+		cli:               common.NewAuthorizedGraphQLClient(bearerToken),
+	}, nil
+}
+
+func (tc *TestContext) RunOperation(ctx context.Context, req *gcli.Request, resp interface{}) error {
+	tnt := os.Getenv("DEFAULT_TENANT")
```

Comment:
Maybe this can be refactored to also wrap the `runCustomOperation` func like `RunOperationWithCustomTenant` and `RunOperationWithCustomScopes`